Controller/UpwardControllerFactory.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\UpwardConnector\Controller;
+
+use Magento\UpwardConnector\Helper\Data;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\ObjectManagerInterface;
+use Magento\Upward\Controller as UpwardController;
+
+class UpwardControllerFactory
+{
+    /**
+     * @var Data
+     */
+    private $helper;
+
+    /**
+     * @var DeploymentConfig
+     */
+    private $deploymentConfig;
+
+    /**
+     * @param ObjectManagerInterface $objectManager
+     * @param Data $configHelper
+     */
+    public function __construct(ObjectManagerInterface $objectManager, Data $configHelper)
+    {
+        $this->objectManager = $objectManager;
+        $this->helper = $configHelper;
```

Comment:
Discussed in Slack, but I think it would be best to have a direct dependency on `ScopeConfigInterface`, and remove the overhead of this AbstractHelper.